osquery/tables/system/windows/services.cpp

```diff
@@ -0,0 +1,144 @@
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Winsvc.h>
+#include <string>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#pragma comment(lib, "Advapi32.lib")
+
+namespace osquery {
+namespace tables {
+
+const std::string kSvcStartType[] = {
+    "BOOT_START", "SYSTEM_START", "AUTO_START", "DEMAND_START", "DISABLED"};
+
+const std::string kSvcStatus[] = {"UNKNOWN",
+                                  "STOPPED",
+                                  "START_PENDING",
+                                  "STOP_PENDING",
+                                  "RUNNING",
+                                  "CONTINUE_PENDING",
+                                  "PAUSE_PENDING",
+                                  "PAUSED"};
+
+const std::map<int, std::string> kServiceType = {
+    {0x00000010, "OWN_PROCESS"},
+    {0x00000020, "SHARE_PROCESS"},
+    {0x00000100, "INTERACTIVE_PROCESS"},
+    {0x00000110, "OWN_PROCESS   (Interactive)"},
+    {0x00000120, "SHARE_PROCESS (Interactive)"}};
+
+SC_HANDLE schSCManager;
+
+BOOL QuerySvcInfo(ENUM_SERVICE_STATUS_PROCESS& svc, Row& r) {
+  SC_HANDLE schService;
+  LPQUERY_SERVICE_CONFIG lpsc = nullptr;
+  LPSERVICE_DESCRIPTION lpsd = nullptr;
+  DWORD cbBufSize = 0;
+
+  schService =
+      OpenService(schSCManager, svc.lpServiceName, SERVICE_QUERY_CONFIG);
+
+  if (schService == NULL) {
+    TLOG << "OpenService failed (" << GetLastError() << ")";
+    return FALSE;
+  }
+
+  (void)QueryServiceConfig(schService, NULL, 0, &cbBufSize);
+  lpsc = (LPQUERY_SERVICE_CONFIG)LocalAlloc(LMEM_FIXED, cbBufSize);
```

Comment:
Actually i tried to understand the difference between mem allocation functions but didn't get satisfied results So, please if you can summarize malloc/free vs. LocalAlloc/LocalFree in its usage cases?

Thanks